python/paddle/utils/cpp_extension/cpp_extension.py

```diff
@@ -367,11 +368,14 @@ def finalize_options(self):
             self.build_lib = self.output_dir
 
     def build_extensions(self):
+        if OS_NAME.startswith("darwin"):
```

Comment:
why only need to check mac compiler

Reply:
Because `self.compiler` will use `gcc` on some mac devices.